oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/config/annotation/web/configurers/OAuth2AuthorizationCodeGrantTests.java

```diff
@@ -355,7 +356,8 @@ private OAuth2AccessTokenResponse assertTokenRequestReturnsAccessTokenResponse(R
 			OAuth2Authorization authorization, String tokenEndpointUri) throws Exception {
 		MvcResult mvcResult = this.mvc.perform(post(tokenEndpointUri)
 				.params(getTokenRequestParameters(registeredClient, authorization))
-				.header(HttpHeaders.AUTHORIZATION, getAuthorizationHeader(registeredClient)))
+				.header(HttpHeaders.AUTHORIZATION, getAuthorizationHeader(registeredClient))
+						.contentType(MediaType.APPLICATION_FORM_URLENCODED_VALUE))
```

Comment:
Please remove all changes where the content type is set in the request as it's redundant. Let's keep the changes in this PR to only what's required for the fix.

Reply:
Reverted all changes for the content type changes.